app/assets/javascripts/application.js

```diff
@@ -42,18 +42,22 @@ $( document ).on('turbolinks:load', function() {
 	});
 
 	// Check if filter needs to be applied
-	if ((window._search != undefined) && (window._search.length > 0 )) {
+	if ((window._search !== undefined) && (window._search.length > 0 )) {
 		// Apply the filter param to the filter box.
 		table.search(window._search);
 		table.draw();
 
 		// Clear the filter in the URL when the search param is updated.
 		table.on( 'search.dt', function () {
-			if ((table.search() != window._search) && (window._search.length > 0 )) {
-				var url = window.location.href;
-				var tail = url.substring(url.lastIndexOf('/') + 1);
-				var clean_tail = tail.substring(0, tail.lastIndexOf('?'));
-				var clean_title = document.title.replace(/#(.*?)\s/, '');
+			if ((table.search() !== window._search) && (window._search.length > 0 )) {
+				const url = window.location.href;
+				const tail = url.substring(url.lastIndexOf('/') + 1);
+				
+				// Remove any query parameters from the segment
+				const clean_tail = tail.substring(0, tail.lastIndexOf('?'));
```

Comment:
It looks like the whole purpose of these lines are to clean up the url. In this case we should wrap it into one function. Please, take this code as an example only:
```suggestion
...
const url_tail_without_parameters = url => {
  const tail = url.substring(url.lastIndexOf('/') + 1);
				
  // Remove any query parameters from the segment
  return tail.substring(0, tail.lastIndexOf('?'));
}
...
  const clean_tail = url_tail_without_parameters(window.location.href);
```